src/test/clojure/clara/test_rules.clj

```diff
@@ -5025,3 +5029,31 @@
     (is (= (query end-session-equal-facts lousy-weather-query)
            (query end-session-identical-facts lousy-weather-query)
            [{}]))))
+
+(deftest test-alpha-batching-with-multiple-ancestors
+  ;; see issue 257 for details of the bug surronding this test
+  (let [accum-state (atom #{})
+        
+        stateful-accum (acc/accum
+                         {:initial-value []
+                          :reduce-fn conj
+                          :retract-fn (fn [items retracted] (remove #{retracted} items))
+                          :convert-return-fn (fn [items] 
+                                               (do 
+                                                 (swap! accum-state conj items)
+                                                 items))})
+        
+        map-rule (dsl/parse-rule [[?maps <- stateful-accum :from [List]]]
+                                 ;; don't care about whats inserted
+                                 (->LousyWeather))
+        
+        identity-list (ArrayList.)
+        linked-list (LinkedList.)
+        
+        ses (-> (mk-session [map-rule])
+              (insert-all [identity-list linked-list])
+              (fire-rules))]
+    
+    (is (and (not (contains? @accum-state [identity-list]))
+             (not (contains? @accum-state [linked-list])))
```

Comment:
I think we'll probably want to do something like

````
(let [accum-state (atom [])]
    (is (not-any? (fn [acc-value] (= (count acc-value) 1)))))
````
instead of this approach using a set and contains? checks.

My understanding is that Clojure considers the behavior of these sort of operations combining Clojure and Java collections to be non-guaranteed.  In this particular case, I don't see a lot of reasons to expect it to break but it would be better to just avoid the issue entirely IMO.

I'm looking at the discussion on http://dev.clojure.org/jira/browse/CLJ-1372 particular the comment that "Rich says "all bets should be off for hasheq/equiv of non-values" where Java collections obviously fall into the class of "non-values"."